src/Illuminate/Validation/Concerns/ValidatesAttributes.php

```diff
@@ -883,19 +883,19 @@ public function validateGt($attribute, $value, $parameters)
     {
         $this->requireParameterCount(1, $parameters, 'gt');
 
-        $comparedToValue = $this->getValue($parameters[0]);
+        $comparedToValue = $this->getValue($parameters[0] ?? '');
 
         $this->shouldBeNumeric($attribute, 'Gt');
 
         if (is_null($comparedToValue) && (is_numeric($value) && is_numeric($parameters[0]))) {
             return $this->getSize($attribute, $value) > $parameters[0];
         }
 
-        if (! $this->isSameType($value, $comparedToValue)) {
-            return false;
+        if ($this->isSameType($attribute, $value, $comparedToValue ?? $parameters[0])) {
+            return $this->getSize($attribute, $value) > $this->getSize($attribute, $comparedToValue ?? $parameters[0]);
         }
 
-        return $this->getSize($attribute, $value) > $this->getSize($attribute, $comparedToValue);
+        return true;
```

Comment:
Why does this return `true` if `isSameType` is false?

Reply:
If `isSameType` is `false`, then **that** method already added a failure message to the bag stating that the field being compared must be of the correct type. 

So if we returned `false` after that point, there would be another failure message added for the `gt`, `gte`, `lt` or `lte` validation. But that's part of the [original issue](https://github.com/laravel/framework/issues/28258) whereas the validation messages returned when the fields under comparison are of different types are often confusing or incorrect. 

So in essence the `return true` statement is just there to bail from the current validation method as the comparison cannot be done with the two different types. I admit it feels somewhat like a hack. And I thought about making the docblock show a mixed return type and then just returning null at the end. But, if its return value is anything but `true` it will add a validation error. 

Before some recent changes, the `isSameType` method used to throw an Exception to kill the process. But apparently some devs (including myself) felt that this was too drastic and preferred to just have the method fail validation. However in its current state, the method will cause the validation to fail, but it doesn't provide any feedback that the reason it failed was because the fields being compared were of different types.

Reply:
the ```validateBail()`` method is a more expressive way to return true when needing to bail. Haven't looked at the code so it may not even make sense to use it here, but it's in the same trait and returns true.

Reply:
@devcircus Good call. I didn't realize that method was available.